datafusion/physical-optimizer/Cargo.toml

```diff
@@ -43,6 +43,7 @@ recursive_protection = ["dep:recursive"]
 [dependencies]
 arrow = { workspace = true }
 datafusion-common = { workspace = true }
+datafusion-datasource = { workspace = true }
```

Comment:
Do we really need to add this new dependncy? I think if you made the optimizer tests look more like the other optimizzer tests this wouldn't be needed

Reply:
Thanks for good point, currently the optimizer will use datasource to pushdown, i will look back more.